Front end

Jon-D-Smith opened the pull request:
Pull Request Changes
 - Created a component for Donor, Pledge, and Prospect
 - Populated pages for Donor, Pledge and Prospect
 - Created a GridWrapper to handle grid styling

Jon-D-Smith committed:
Created Pledge and Prospect Components/Tested on respective pages

Jon-D-Smith committed:
Linter fixes

Jon-D-Smith committed:
Linter changes

Jon-D-Smith committed:
Fixed linter issues

Jon-D-Smith committed:
Added donor page, donor comp, and gridwrapper util

Jon-D-Smith committed:
Cleaned up pages

Jon-D-Smith committed:
Formatting Issue for notes header

matthew-r-clark commented on src/utils/GridWrapper.js:
```diff
@@ -0,0 +1,17 @@
+import styled from '@emotion/styled';
+
+const GridWrapper = ({ children, cols }) => {
```
Should we go ahead and add a `rows` prop as well and default both to `1`?

Jon-D-Smith replied:
I was thinking about that. We should add rows as an option, but default it to auto. Often times with grid, you don't need to set a finite number of rows, and if we can leave it as an optional component I think that would be nice.

matthew-r-clark replied:
Sounds good to me.

matthew-r-clark commented on src/utils/GridWrapper.js:
```diff
@@ -0,0 +1,17 @@
+import styled from '@emotion/styled';
+
+const GridWrapper = ({ children, cols }) => {
```
Also I'd be okay with just calling the component `Grid` to keep it more concise.

matthew-r-clark replied:
You could just create the component similar to this, using a function to change style based on props and then export it. That way you can just export the base `Grid` component you've got there without having to create a `GridWrapper` component.
https://emotion.sh/docs/styled#changing-based-on-props

matthew-r-clark replied:
Sorry if what I said isn't super clear. I can try to explain it better later if necessary.

Jon-D-Smith replied:
It was clear! I just made the changes. It is definitely a lot cleaner! Let me know if you see anything else!

matthew-r-clark commented:
Sounds good to me.

On Thu, Jul 8, 2021, 2:18 PM Jon Smith ***@***.***> wrote:

> *@Jon-D-Smith* commented on this pull request.
> ------------------------------
>
> In src/utils/GridWrapper.js
> <https://github.com/matthew-r-clark/drm/pull/16#discussion_r666457059>:
>
> > @@ -0,0 +1,17 @@
> +import styled from ***@***.***/styled';
> +
> +const GridWrapper = ({ children, cols }) => {
>
> I was thinking about that. We should add rows as an option, but default it
> to auto. Often times with grid, you don't need to set a finite number of
> rows, and if we can leave it as an optional component I think that would be
> nice.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/matthew-r-clark/drm/pull/16#discussion_r666457059>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJGIETALWIIDDSX6UI7Q75TTWX2X3ANCNFSM475NNS2A>
> .
>

Jon-D-Smith committed:
Changed GridWrapper component to a new Grid component

matthew-r-clark commented:
So earlier I was just kind of glancing at this PR during breaks or whatever. I finally sat down and looked through it a little more carefully.

I've been thinking more about the `Donor`, `Prospect`, and `Pledge` components and I think they need to just be defined in the page component they're used within, rather than imported as a separate module. For one, I think it's a little unclear what the components are for if you come across one by itself, and you would have to find where it's used to understand. Second, they're each a specialized component for list items on a single page so there's definitely not going to be any reuse. And ultimately, they are just laying out the structure of the content in those rows, which would make the most sense to see within the context of the page it's going to be rendered on.

Sorry about changing my mind on this :sweat_smile:

matthew-r-clark commented on src/pages/[userId]/pledges.js:
```diff
@@ -1,7 +1,46 @@
 import { useRouter } from 'next/router';
 
+import Grid from '../../utils/Grid';
+
```
You can just put all the import statements in a group at the top. You only need an empty line between the last import and the first line of code following. There may be some exceptions to this as well.

matthew-r-clark commented on src/pages/[userId]/pledges.js:
```diff
@@ -1,7 +1,46 @@
+    },
+  ];
+
+  const pledges = pledgesArr.map((pledge) => <Pledge key={pledge.id} donor={pledge} />);
```
Since this is so concise already, we may just want to put it right into the markup. It may get a little more complicated later though when we start adding more functionality with the pledges, but we can deal with it at that point.

Also after thinking about it more I think we should use the prop name `pledges` here, and do the same on the other pages with their respective names. In addition we are going to need to change our language around donors. Dani was telling me that FOCUS is trying to use the term "ministry partner" instead. The unfortunate thing is it's so many more characters lol. But it's helpful to have our code reflect the actual names so it will be worth making the change.

matthew-r-clark commented on src/pages/[userId]/prospects.js:
```diff
@@ -1,7 +1,51 @@
 import { useRouter } from 'next/router';
 
```
Remove blank lines between imports.

matthew-r-clark commented on src/pages/[userId]/prospects.js:
```diff
@@ -1,7 +1,51 @@
+    },
+  ];
+
+  const prospects = prosArr.map((prospect) => <Prospect key={prospect.id} donor={prospect} />);
```
Change prop name `donor` to `prospects`.

Also lets avoid abbreviating variable names, like `prosArr` because it can easily get confusing later on. One way to get around this naming issue would be to:
- name the array of prospects data `prospects` instead of `prosArr` (we'll do the same thing when we're actually retrieving this from the db)
- like my comment on the pledges page, we can move the array of `Prospect` components created on line 38 straight into the markup on line 48. So instead of `{pledges}` we have `prospects.map((prospect) => <Prospect key={prospect.id} donor={prospect} />)`. Then we avoid needing to create another variable name.

matthew-r-clark commented on src/pages/donors.js:
```diff
@@ -1,3 +1,41 @@
+import Donor from '../components/Donor';
+
```
Remove blank line.

matthew-r-clark commented on src/pages/donors.js:
```diff
@@ -1,3 +1,41 @@
+    },
+  ];
+
+  const donors = donorArr.map((donor) => <Donor key={donor.id} donor={donor} />);
```
Rename `donors` to `ministryPartners`, and `donor` to `ministryPartner`.

Move the map statement into the markup.

Jon-D-Smith committed:
Refactored code per Pull Request comments

Jon-D-Smith commented:
I believe I have fixed all of the issues in this Pull Request. I renamed the donor page to ministry-partners.js and linked the Menu to that HREF. I set the text to partners because Ministry Partners was too long without changing the CSS causing it to overlap the main page. I figure we can change this when we want to deal with the css.

Jon-D-Smith committed:
Fixed pledges Array name

matthew-r-clark commented:
Looks great man. Good work!
